it/lessons/advanced/concurrency.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: page
+title: Concorrenza
+category: advanced
+order: 4
+lang: it
+---
+
+Uno dei punti di forza di Elixir è il suo supporto alla concorrenza. Grazie alla ErlangVM, scrivere programmi concorrenti in Elixir è più semplice di quanto uno possa aspettarsi. Il modello di concorreza si basa sugli Attori, processi indipendenti che comunicano con altri processi, attraverso il passaggio di messaggi.
+
+In questa lezione daremo un'occhiata ai moduli sulla concorrenza che vengono provveduti con l'installazione di Elixir. Nel prossimo capitolo, ci occupiamo dei _behaviour OTP_ che ci permettono di scrivere programmi concorrenti.
+
+## Tavola dei Contenuti
+
+- [Processi](#processi)
+  - [Message Passing](#message-passing)
+  - [Process Linking](#process-linking)
+  - [Process Monitoring](#process-monitoring)
+- [Agenti](#agenti)
+- [Task](#task)
+
+## Processi
+
+I processi che vivono nella ErlangVM sono leggeri, e possono tenere occupate tutte le CPU del sistema. Nonostante questi possano sembrare thread nativi, i processi Erlang sono più semplici, e non è raro trovare migliaia di processi concorrenti in un'applicazione Elixir.
+
+Il modo più semplice per lanciare un processo è attraverso il comando `spawn`, che accetta una funzione anonima o pre-definita. Quando creiamo un processo, il suo _Process Identifier_, o PID, ci viene restituito; con esso, possiamo identificare unicamente il processo creato all'interno della nostra applicazione.
+
+Per iniziare, creiamo un modulo e definiamo una funzione che vorremmo lanciare:
+
+```elixir
+defmodule Example do
+  def add(a, b) do
+    IO.puts(a + b)
+  end
+end
+
+iex> Example.add(2, 3)
+5
+:ok
+```
+
+Per lanciare la stessa funzione in modo asincrono, usiamo `spawn/3`:
+
+```elixir
+iex> spawn(Example, :add, [2, 3])
+5
+#PID<0.80.0>
+```
+
+### Message Passing
+
+Per comunicare, i processi usano _message passing_. A riguardo, ritroviamo due componenti principali: `send/2` e `receive/1`. La funzione `send/2` ci permette di mandare message ad altri PID; per ascoltare per messaggi in arrivo, usiamo `receive/1` ed (opzionalmente) eseguiamo un _match_ sui messaggi in arrivo. Se un messaggio non soddisfa nessun _match_, il processo ignora il messaggio e continua ininterrotto.
```

Comment:
I processi utilizzano lo scambio di messaggi come metodo di comunicazione, facendo uso di due componenti principali: `send/2` e `receive/1`.
La funzione `send/2` ci permette di mandare messaggi ad altri PID, mentre `receive/1` ci consente di ascoltare quelli in arrivo effettuando un confronto (_match_) opzionale per determinare se il messaggio va considerato o ignorato.